opendevin/core/config.py

```diff
@@ -79,13 +115,53 @@ class LLMConfig:
     output_cost_per_token: float | None = None
     ollama_base_url: str | None = None
     drop_params: bool | None = None
+    memory_summarization_fraction: float = 0.75
+
+    @classmethod
+    def load_from_env(cls, env_dict: dict[str, str]) -> 'LLMConfig':
+        config = cls()
+        for f in fields(cls):
+            env_var_name = f'LLM_{f.name.upper()}'
+            if env_var_name in env_dict:
+                value = env_dict[env_var_name]
+                if value:
+                    setattr(config, f.name, cls._cast_value(f.type, value))
+        return config
+
+    @classmethod
+    def load_from_toml(cls, toml_config: dict) -> dict[str, 'LLMConfig']:
+        llm_configs = {}
+        default_config = cls()
+
+        # Load default [llm] section if it exists
+        if 'llm' in toml_config and isinstance(toml_config['llm'], dict):
+            default_config = cls(**toml_config['llm'])
+            llm_configs['llm'] = default_config
+
+        # Load custom LLM configs, falling back to default for unspecified attributes
+        for key, value in toml_config.get('llm', {}).items():
```

Comment:
This worries me that pulling values from one section into another introduces unexpected behaviors.
Sections should be standalone and not treated as inheritable classes, so to speak.
If a required value is missing, there just needs to be a clear error for the user to see.